build.bash

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+nproc_default=4
+
+script_name=$(basename "${0}")
+
+show_help() {
+    cat << EOF
+Usage: ./$script_name [OPTIONS]
+
+Build script wrapper around CMake. Supplied arguments that do not match the
+options below will be passed to CMake when generating the build system.
+
+Options:
+      --clean   Delete build directory before invoking CMake.
+      --mpi     Compile MPI executable.
+  -h, --help    Show this screen.
+
+Enabling debug mode:
+
+  The release build is default. To enable debug mode, specify the CMake option
+  -DCMAKE_BUILD_TYPE=Debug when invoking $script_name.
+
+Enabling verbose output from Makefile builds:
+
+  To enable more verbose output from Makefile builds, specify the CMake option
+  -DCMAKE_VERBOSE_MAKEFILE=ON when invoking $script_name.
+
+Parallel compilation:
+
+  By default, the number of parallel jobs used in the compilation is
+  $nproc_default. This value can be overwritten by setting the environment
+  variable CMAKE_BUILD_PARALLEL_LEVEL.
+
```

Comment:
Can we add this to the main documentation? It can be a note box under installation.md with something like:
The compilation is done in parallel by default. See `build.bash -h` for more information on the default number of parallel jobs and how to customise it.

I think it's important that the default behaviour is clearly stated in the doc.